lib/galaxy/tools/__init__.py

```diff
@@ -473,8 +480,11 @@ def parse( self, tool_source, guid=None ):
 
         self.version = tool_source.parse_version()
         if not self.version:
-            # For backward compatibility, some tools may not have versions yet.
-            self.version = "1.0.0"
+            if self.legacy_defaults:
+                # For backward compatibility, some tools may not have versions yet.
+                self.version = "1.0.0"
+            else:
+                raise Exception( "Missing tool version.")
```

Comment:
Did a pass at this - when I push this branch again it will add more data to all these exceptions.